pyccel/parser/semantic.py

```diff
@@ -2315,12 +2314,22 @@ def _visit_FunctionDef(self, expr, **settings):
         is_private   = expr.is_private
 
         header = expr.header
+        args_number = len(expr.arguments)
         if header is None:
             if cls_name:
                 header = self.get_header(cls_name +'.'+ name)
+                args_number -= 1
             else:
                 header = self.get_header(name)
 
+        if header:
+            if (args_number != len(header.dtypes)):
+                msg = 'The number of arguments in the function {} ({}) does not match the number of types in decorator/header ({}).'.format(name ,args_number, len(header.dtypes))
+                if (args_number < len(header.dtypes)):
+                    errors.report(msg, symbol=expr.arguments, severity='warning')
+                else:
+                    errors.report(msg, symbol=expr.arguments, severity='fatal')
```

Comment:
Could you please explain why do you raise only a warning if the number of function arguments is smaller than what given in the decorator/header? I understand that in this case the decorator/header contains enough information to correctly describe all function arguments, but in practice this situation will typically happen when the user modifies an existing function but forgets to update its decorator/header...

Reply:
i don't think it good to stop Pyccel from completing the job in this situation, if we raise an error the user will need to modify the code and then execute Pyccel again, given that the types in the header/decorator goes respectively with the function arguments i think that a warning is enough to pay the attention of the user, and if we raise an error it will result a bad user experience.

Reply:
OK, I can agree with that.